### PR TITLE
Link Aliasing Beta Documentation

### PR DESCRIPTION
**Pull request by gsinghAppboy**
# Pull Request/Issue Resolution

**Description of Change:**
Hidden documentation for an upcoming feature that will be shared for a few customers (Link Aliasing)

Closes #**ISSUE_NUMBER_HERE**

### Is this change associated with a Braze feature/product release?
- [ ] Yes (__Insert Feature Release Date Here__)
- [ ] No

> If yes, please note the date of the feature release.

---
---

<details>
<summary>PR Checklist</summary>

- [ ] Check that all links work!
- [ ] Ensure you have completed [our CLA](https://www.braze.com/docs/cla/).
- [ ] Tag @Timothy-Kim and @KellieHawks as a reviewer when your work is _done and ready to be reviewed for merge_.
- [ ] Tag others as Reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `config/nginx.conf.erb`.

</details>

<details>
<summary>Writing & Style Quick Tips</summary>

- Use the formatting shown on the [Styling Test Page](https://github.com/Appboy/braze-docs/blob/develop/_docs/_home/styling_test_page.md) to create unique templates and vehicles for your information.
- Use Action verbs!
- Capitalize proper nouns, unique Braze tools and features, custom Braze channels, and all Header titles.
- Redirect when necessary!
- Always speak to the Braze customer - refer to them as “you”, whether they are an Engineer/Developer, a Marketer, or an Admin.
- On Voice: You should aim to be a Friendly Professor. Imagine, if you will, the smartest, but also kindest professor you had. You know, the one who showed you your potential and tested you, but also went slowly in class so you could take great notes and learn some things? Don't be too peppy, but don't be formal. You’re not selling anything - you’re instructing and informing and enabling.

_Consult the [Docs Text Formatting Guide](https://github.com/Appboy/braze-docs/wiki/Special-Formatting) and the [Docs Writing Style Guide & Best Practices](https://github.com/Appboy/braze-docs/wiki/Writing-Style-Guide-&-Best-Practices) for more details._

</details>

<details>
<summary>ATTENTION: For PR Reviewers</summary>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Preview all changes in the linked Heroku environment (click `View deployment` button below, then "Docs". A `502` error just means you should refresh until you see the Docs Home page.
</details>

---
---

### CODE REVIEW

**Comment by gsinghAppboy** [comment]
@KellieHawks @willng-braze adding in doc for a new feature.  It will be shared with customers during this beta process.

File: _docs/_hidden/private_betas/link_aliasing.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+nav_title: Link Aliasing
+permalink: /linkaliasing/
+description: "This article describes how Link Aliasing works and some of the nuances with the feature."
+hidden: true
+---
+
+ 
+### Link Aliasing
+Goal: Identifying links set in an email message from Braze by a user generated friendly name.
+Goal: Having the ability to re-target users based on clicking a specific link sent in an Email.
```
Comment (willng-braze) [inline _docs/_hidden/private_betas/link_aliasing.md:11]: Do we need to mention trigger action?

File: _docs/_hidden/private_betas/link_aliasing.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+---
+
+ 
+### Link Aliasing
+Goal: Identifying links set in an email message from Braze by a user generated friendly name.
+Goal: Having the ability to re-target users based on clicking a specific link sent in an Email.  
+ 
+ 
+#### How Link Alisaing works
+Link Aliasing works by decorating a Braze generated query parameter on links in the email channel.  For each known link that is present in the Email body, Braze will add a ‘lid={{somevalue}}’ when the client goes to the ‘Link Management’ section or hits ‘done’ on the editor (returning to the ‘compose’ section in campaigns or step overview in canvas).
```
Comment (willng-braze) [inline _docs/_hidden/private_betas/link_aliasing.md:15]: {{somevalue}} seem to suggest its liquid.  Maybe [0-9a-f]{12} is a more universal language.  Not sure what's the best lingo

File: _docs/_hidden/private_betas/link_aliasing.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ 
+ 
+ 
+#### Feature enablement
+The process to enable link aliasing is simple and does not require any downtime.  This enablement is not backward compatible, meaning that previously created messages or content blocks will not be recognized by this feature (unless they are modified and launched again).  
+
+##### Preconditions / Limitations
+Client is okay with their message being modified by an HTML parser.  This could lead to the parser ‘correcting’ potentially incorrect HTML. (currently this already happening if you use features such as pre-header input field, liquid statements, or link templates)
+Client is okay with the implications of being in a partially migrated state (some messages will have aliasing, some will not (same with content blocks))
+Editing messages or content blocks prior to having the feature enabled will result in Braze editing links
```
Comment (willng-braze) [inline _docs/_hidden/private_betas/link_aliasing.md:35]: Only messages not CBs

File: _docs/_hidden/private_betas/link_aliasing.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+Link Aliasing is not supported for modifying links in 'mso commented elements'
```
Comment (willng-braze) [inline _docs/_hidden/private_betas/link_aliasing.md:36]: I think the inverse is more appropriate.

Link alias is only supported in href attributes within HTML anchor tags.

File: _docs/_hidden/private_betas/link_aliasing.md (hunk 3 above)
Comment (willng-braze) [reply _docs/_hidden/private_betas/link_aliasing.md:35]: Also since links have changed, user should reapply link templates as they see fit.

**Review by willng-braze — APPROVED** [review]
Left a few comments, this looks good!  I assume this will evolve overtime as new features are added and as functionality is tweaked.

File: _docs/_hidden/private_betas/link_aliasing.md (hunk 3 above)
Comment (gsinghAppboy) [reply _docs/_hidden/private_betas/link_aliasing.md:35]: what do you mean only messages? if you edit an 'old cb', it will get modified too

Comment (willng-braze) [reply _docs/_hidden/private_betas/link_aliasing.md:36]: Not for partial migration.  Partial migration only marks up CBs created after migration time.

**Review by KellieHawks — APPROVED** [review]
Looks great! All three docs have been reviewed by the docs team.

**Pull request merged**